### PR TITLE
Expose py_recipe::global_properties

### PR DESCRIPTION
**Pull request by noraabiakar**
* Rename `pyrecipe::get_probes` to `py_recipe::probes`
* Remove `global_properties_shim` : It's easy to make mistakes with the class, as it holds a `mechanism_catalogue` and a `cable_cell_global_properties` which holds a pointer to the catalogue. This would likely have caused issues with the users.
* Expose `py_recipe::global_properties`

### CODE REVIEW

File: python/single_cell_model.cpp [hunk 1]
```diff
@@ -270,6 +270,7 @@ void register_single_cell(pybind11::module& m) {
                 return m.traces();},
             "Holds sample traces after a call to run().")
         .def_readwrite("properties", &single_cell_model::gprop, "Global properties.")
+        .def_readwrite("catalogue", &single_cell_model::cat, "Mechanism catalogue.")
```
Comment (bcumming) [inline python/single_cell_model.cpp:273]: The global properties hold a pointer to a catalogue.
At the moment this pointer is only set to point to the `single_cell_model::cat` when `::run` is called, so there will be a mismatch with the catalogue in the properties.

Instead, can users just set the catalogue by accessing `properties`, i.e:
```Python
model.properties.catalogue = ...;
```

**Review by bcumming — CHANGES_REQUESTED** [review]
A few small nits.

File: python/cells.cpp [hunk 2]
```diff
@@ -60,6 +60,15 @@ arb::util::unique_any convert_cell(pybind11::object o) {
                       + "\" which does not describe a known Arbor cell type");
 }
 
+// Convert global properties inside a Python object to a
+// std::any, as required by the recipe interface.
+std::any convert_gprop(pybind11::object o) {
```
[the remark this answers is not in the document]
Comment (noraabiakar) [reply python/cells.cpp:65]: Right, I'll add that fix here too.

File: python/single_cell_model.cpp (hunk 1 above)
Comment (noraabiakar) [reply python/single_cell_model.cpp:273]: Yes, that's true. But we don't expose the `catalogue` of the global properties to the user any more. So `model.properties.catalogue` won't work. Instead we have `model.properties.register(cat)` which will point the global properties catalogue to `cat`.

**Review by bcumming — APPROVED** [review]

**Pull request merged**